src/shims/unix/socket.rs

```diff
@@ -19,7 +21,11 @@ struct SocketPair {
     // gone, and trigger EPIPE as appropriate.
     writebuf: Weak<RefCell<Buffer>>,
     readbuf: Rc<RefCell<Buffer>>,
+    /// When a socketpair instance is created, two socketpair file descriptions are generated.
+    /// The peer_fd field holds a weak reference to the file description of peer socketpair.
+    peer_fd: WeakFileDescriptionRef,
```

Comment:
So `writebuf` is always the same as `peer_fd.downcast::<SocketPair>().unwrap().readbuf`, right? Seems like it's possible to remove `writebuf` and even remove the `RefCell` around `Buffer`, then... but maybe in a future PR. For now, just leave a TODO to document this possibility.